docker/docker-compose.yml

```
@@ -18,11 +18,14 @@ services:
       - POSTGRES_HOST=postgres
       - POSTGRES_DATABASE=automatisch
       - POSTGRES_USERNAME=automatisch_user
+      - POSTGRES_PASSWORD=automatisch_password
+      - ENCRYPTION_KEY=automatisch_encryption_key
+      - APP_SECRET_KEY=automatisch_app_secret_key
```

If we provide default values for this and the following environment variable, we will not generate a random key with `openssl` in the entry point script later. I think we want to just pass these environment variables through if provided.

```suggestion
      - ENCRYPTION_KEY
      - APP_SECRET_KEY
```

This would be my suggestion.

ref: https://docs.docker.com/compose/environment-variables/#pass-environment-variables-to-containers

---

yes the `:-` notation in the entrypoint script will use the provided value if it's not already defined. I went with randomly generated instead of a static default which is publicly known in this repo.

---

I get your point here. However, we would like to avoid this as people may unintentionally stick to default values, which are not secure by default.

We would like to leave the environment variables without values that we can either pass the given values in runtime or auto-generate values with `openssl`.

---

This should meet those requirements. Are you talking about the entrypoint script or the compose file?

---

I'm talking about the compose file. If I do not provide environment variables while running `docker compose up`, it uses the default values in the docker-compose.yaml file. Therefore, `openssl` command never generates values to override them.

I'm writing this based on experience on my local when I run `docker compose up` on a fresh setup. Does it anyhow work differently on your end? By the way, there are also some other relevant remarks on this matter in the last review I submitted.

---

Ah I see what you mean. Those are meant to be placeholders to be changed by the user. I could make it more clear, but this could also be mentioned in documentation. Should I leave it blank then?

---

Exactly. We are concerned that the placeholder values may be left as-is and, thus, not secure by default. It's better to leave them empty and auto-generate those values (in the entrypoint script) if not provided.

---

I guess the same applies to the db user and password as well. I think it should just be clearly marked to be replaced like 

```
  - POSTGRES_USERNAME=<USER> # replace with your own user
  - POSTGRES_PASSWORD=<PASSWORD> # replace with your own password
  - ENCRYPTION_KEY=<ENCRYPTION_KEY> # replace with your own key
  - APP_SECRET_KEY=<APP_SECRET_KEY> # replace with your own key
```

---

I have pushed a commit to nullify the `ENCRYPTION_KEY` and `APP_SECRET_KEY` environment variables in the docker-compose.yml. I suggest we leave Postgres's defaults in place as we want to make `docker compose up` as convenient as possible for first-time triers.

I will resolve this thread. If you have a remark, let's initiate a new thread. @ksurl